### DISCUSSION TITLE
Variation in Mutation Rates

### DISCUSSION THREAD

**Discussion by David-Peede** [tskit-dev/msprime#2117] [Q&A] [answered]
Hello all, I know one can specify variable mutation rates throughout time and throughout the genome, but I was wondering if anyone knew if it is possible or of any tricks/workarounds to simulate variable mutation rates between populations.

**Comment by jeromekelleher** [comment, marked as answer]
If the populations are independent, I guess you could run separate simulations with the different mutation rates and then (with a little work - I can help with this if it's needed) merge the two results into a single tree sequence. You could then simulate the deep history at the same ancestral mutation rate.

If you need gene flow between the populations, the I can't see how it would be done.

**Reply by David-Peede** [reply]
I see. Is there someone who has done something analogous to the merging you are describing that I could read up on—I have never thought about merging results into the same tree sequence before! Also, could you help me gain intuition on why this couldn't be done with gene flow? If you can merge the output from two `sim_mutations()` runs, don't you just need to keep track of the migration and coalescent times to know for which given time interval what mutation rate to use—I naively think this would be analogous to the current set-up where you can have different mutation during different time spans.

**Reply by jeromekelleher** [reply]
You'd have to do something like this:
```python3
tsA = msprime.sim_ancestry(..., end_time=t)
tsA = msprime.sim_mutations(tsA, A_mutation_rate)
tsB = msprime.sim_ancestry(..., end_time=t)
tsB = msprime.sim_mutations(tsB, B_mutation_rate)
tsAB = mush_together(tsA, tsB)
tsABC = msprime.sim_ancestry(initial_state=tsAB, ...)
tsABC = msprime.sim_mutations(tsABC, C_mutation_rate, start_time=t)
```

It's messy though.
